### ISSUE TITLE
QInput mask still freezes browser in some specific cases

### ISSUE THREAD

**Issue by abelsiqueira** [quasarframework/quasar#14558]
### What happened?

We use the input mask `https://orcid.org/####-####-####-###X`. Typing `1234-1234-1234-` will cause the browser to freeze. Looking into other issues, this looks like the same as #4381, but without being at the end of input.

### What did you expect to happen?

I expected all `-` characters to be ignored.

### Reproduction URL

https://jsfiddle.net/abelsiqueira/km82nxo7/

### How to reproduce?

1. Go to https://jsfiddle.net/abelsiqueira/km82nxo7/
2. enter `1234-1234-1234-` (explicitly enter the -).

Alternative in production:
1. Go to https://citation-file-format.github.io/cff-initializer-javascript/#/authors
2. Click "Add author"
3. Go to "orcid" input
4. enter `1234-1234-1234-` (explicitly enter the -).

Alternative to writing the number yourself: copy `1234-1234-1234-` and paste twice.

### Flavour

Vue CLI Plugin (vue-cli-plugin-quasar)

### Areas

Components (quasar)

### Platforms/Browsers

Firefox, Other

### Quasar info output

```Shell
Operating System - Linux(5.19.9-arch1-1) - linux/x64
NodeJs - 16.15.1

Global packages
  NPM - 8.11.0
  yarn - Not installed
  @quasar/cli - undefined
  @quasar/icongenie - Not installed
  cordova - Not installed

Important local packages
  quasar - 2.0.3 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app - 3.0.4 -- Quasar Framework local CLI
  @quasar/extras - 1.10.9 -- Quasar Framework fonts, icons and animations
  eslint-plugin-quasar - Not installed
  vue - 3.1.5 -- The progressive JavaScript framework for buiding modern web UI.
  vue-router - 4.0.10
  vuex - Not installed
  electron - Not installed
  electron-packager - Not installed
  electron-builder - Not installed
  @babel/core - 7.14.8 -- Babel compiler core.
  webpack - 5.48.0 -- Packs CommonJs/AMD modules for the browser. Allows to split your codebase into multiple bundles, which can be loaded on demand. Support loaders to preprocess files, i.e. json, jsx, es7, css, less, ... and your custom stuff.
  webpack-dev-server - 4.0.0-beta.3 -- Serves a webpack app. Updates the browser on changes.
  workbox-webpack-plugin - Not installed
  register-service-worker - 1.7.2 -- Script for registering service worker, with hooks
  typescript - 4.2.2 -- TypeScript is a language for application scale JavaScript development
  @capacitor/core - Not installed
  @capacitor/cli - Not installed
  @capacitor/android - Not installed
  @capacitor/ios - Not installed

Quasar App Extensions
  @quasar/quasar-app-extension-testing - 2.0.0-beta.2 -- A Quasar App Extension for managing Test Harnesses
  @quasar/quasar-app-extension-testing-unit-jest - 3.0.0-alpha.3 -- A Quasar App Extension for running Jest tests
```

### Additional context

Also Opera browser.

**Comment by rstoenescu** [comment]
Fix will be available in Quasar v2.9.2 and v1.21.1

**Comment by abelsiqueira** [comment]
Thanks!

**Comment by abelsiqueira** [comment]
Hello again, thanks for fixing the initial issue, but apparently the second use case is not fixed (copy-pasting). Tested with 2.10.0. Should I create a new issue?

**Comment by pdanpdan** [comment]
No, I'll use this one

**Comment by abelsiqueira** [comment]
Thanks!

**Comment by rstoenescu** [comment]
Fix will be available in Quasar 2.10.1 & 1.22.1
